### ISSUE TITLE
Sending an IpcReceiver over a previously sent IpcChannel doesn't work on Windows

### ISSUE THREAD

**Issue by zaddach** [servo/ipc-channel#310]
I'm trying to set up a secondary channel through which I can transfer Senders/Receivers/SharedMemory. Since the secondary channel has been created in a single process, both its server and client named pipe have the same PID. When the secondary channel's pipe handle gets transferred to the other process (via DuplicateHandle), the PID for its server end isn't updated. When another handle (e.g., from a shared memory blob) is now sent over the connection, the transfer will fail on the other end, as the PID encoded in the OutOfBand message is wrong (and the handle hasn't been duplicated into the receiver's process).

- [The code for getting the server PID is here](https://github.com/servo/ipc-channel/blob/7f432aab1eb77bb2c860881153799087918558e9/src/platform/windows/mod.rs#L1227)
- [The PID check on the receiver side is here](https://github.com/servo/ipc-channel/blob/7f432aab1eb77bb2c860881153799087918558e9/src/platform/windows/mod.rs#L162)
- [Windows' pipefs behavior of setting server/client PID at pipe creation time is documented here](https://googleprojectzero.blogspot.com/2019/09/windows-exploitation-tricks-spoofing.html)

Pseudocode of what I'm trying to do at the sender's side:
```
let sender: IpcSender<Requests> = IpcSender::connect(server_name).unwrap();
let (secondary_sender, secondary_sender_other_end) = channel<SecondaryRequests>().unwrap();
sender.send(Requests::Init(secondary_sender)).unwrap();
let shared_memory = IpcSharedMemory::from_byte(0xba, 1024 * 1024);
// The next command will fail on the receiver side, because the server PID of the secondary OsIpcSender is wrong (still the one of the current process, not the transferred OsIpcReceiver)
secondary_sender.send(SecondaryRequests::Mem(shared_memory)).unwrap();
```

**Comment by wusyong** [comment]
I hit the similar problem when attempting to add multiprocess support for Servo on Windows. It resulted in the same panic in [oob_data](https://github.com/servo/ipc-channel/blob/7f432aab1eb77bb2c860881153799087918558e9/src/platform/windows/mod.rs#L162).

**Comment by wusyong** [comment]
I wonder if we can replace the implementation with [anonymous pipes](https://learn.microsoft.com/en-us/windows/win32/ipc/anonymous-pipe-operations) instead. It states it's safe to duplicate the handle to other process.

**Comment by zaddach** [comment]
Anonymous pipes aren't very popular on Windows because they don't
support overlapped operations. Internally they're a pair of named
pipes anyways, just that the name isn't visible outside the
Windows-internal code.

**Comment by wusyong** [comment]
yeah I saw there are lots of overlapped operations are not supported which ipc-channel needs them.
I wonder how Firefox does it because it also uses named pipe for sure.
